src/main/java/slimeknights/tconstruct/library/modifiers/hook/ranged/BowAmmoModifierHook.java

```diff
@@ -183,7 +183,10 @@ static ItemStack findAmmo(IToolStackView tool, ItemStack bow, Player player, Pre
       if (needed > standardAmmo.getCount()) {
         // consume the whole stack
         resultStack.grow(standardAmmo.getCount());
-        player.getInventory().removeItem(standardAmmo);
+        standardAmmo.setCount(0);
+        if (user instanceof Player player) {
+          player.getInventory().removeItem(standardAmmo);
+        }
```

Comment:
I want to discuss this more; how is a non-player expected to remove the ammo from their inventory? Or are we just assuming they don't care about cleaning up empty stacks?

I know most mobs do infinite ammo, but imagine a dispenser like shooter.

Reply:
I would say the mob needs to be able to clean up empty stacks themselves. What harm will it do to not remove empty stacks? They are empty and `isEmpty()` returns true

Reply:
Mostly just memory waste, which is why vanilla typically does it.

Reply:
I don't think there is a realistic method to clear mob inventory, as each mob would have different implementations. Sounds like it doesn't matter that much.

Though if you think this part should be part of the API, then maybe `LauncherUserInfo` would be a place to add it.